sympy/solvers/diophantine.py

```diff
@@ -875,9 +875,8 @@ def _diop_quadratic(var, coeff, t):
                     + (sqa*g*u**2 + D*u + sqa*F) // _c
 
                 for z0 in range(0, abs(_c)):
-                    if divisible(
-                            sqa*g*z0**2 + D*z0 + sqa*F,
-                            _c):
+                    if (divisible(sqa*g*z0**2 + D*z0 + sqa*F, _c) and
```

Comment:
The value `a*E - sqa*D*w*sqc` is almost `-sqa*_c`. Was this condition mentioned in the reference at the beginning of this section of code? You say something about checking whether `x` and `y` are integers but this test is not really transparent in that regard. Can you explain the derivation of the condition?